# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/barrage/hotfix-refresh.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- barrage-1.0.8.orig/src/main.c	2026-07-26 06:51:46.775753313 +0200
+++ barrage-1.0.8/src/main.c	2026-07-26 06:51:46.778407694 +0200
@@ -301,7 +301,7 @@
 		SDL_FillRect( screen, 0, 0x0 );
 		SDL_SetAlpha( buffer, SDL_SRCALPHA | SDL_RLEACCEL, (int)alpha );
 		SDL_BlitSurface( buffer, 0, screen, 0 );
-		refresh_screen(screen);
+		refresh_screen();
 		if (delay>0) SDL_Delay(10);
 	}
 
@@ -311,7 +311,7 @@
 		SDL_BlitSurface( buffer, 0, screen, 0 );
 	else
 		SDL_FillRect( screen, 0, 0x0 );
-	refresh_screen(screen);
+	refresh_screen();
 	SDL_FreeSurface( buffer );
 }
 
@@ -529,7 +529,7 @@
 			sprintf( buf, "Topgunner #%i", rank+1 );
 			SDL_WriteTextCenter( ft_chart, screen, 320, 230, buf );
 			SDL_WriteTextCenter( ft_chart, screen, 320, 290, "Sign here:" );
-			refresh_screen(screen);
+			refresh_screen();
 			SDL_EnterTextCenter( ft_chart, screen, 320, 320, 18, player_name );
 			chart_add_entry( player_name, player_score );
 		}
@@ -538,7 +538,7 @@
 					ft_chart, screen, 320, 290, "Not enough to be a topgunner!" );
 			SDL_WriteTextCenter( 
 					ft_chart, screen, 320, 320, "Go, try again! Dismissed." );
-			refresh_screen(screen);
+			refresh_screen();
 			wait_for_input();
 		}
 		chart_save();
@@ -635,7 +635,7 @@
 		draw_cursor( screen, x, y );
 		
 		/* udpate screen */
-		refresh_screen(screen);
+		refresh_screen();
 		frames++;
 
 		/* end game? */
